Common/YamlParsers/V2/Factories/ModuleYamlParserInstance.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+
+namespace Common.YamlParsers.V2.Factories
+{
+    public static class ModuleYamlParserInstance
```

Comment:
Remove this class, make ModuleYamlParser constructor private, and add ModuleYamlParser.Instance/Get.

Reply:
> > > What about using V2 parsers instead old?
> > > Remove old parsers, and V2-suffix from new, we do not want to support both versions.
> > > Also old ini-parsers can be removed with `convert-spec` command.
> > 
> > 
> > When you say "old" which ones do you mean? Current ones or those which work with ini-files?
> > Either way, switching to a new version of parsers is a big deal and should be done separately. It probably should require feature-flags mechanism to make the transition smooth.
> 
> Means old = current.
> So now is no some effect of this PR?

Yep, it's a "tech release".

Reply:
I've tried to move initialization logic from `ModuleYamlParserInstance` то `ModuleYamlParser` and it became messy.
It's better to keep business logic and initialization logic separate. 
This way it would be easier to introduce DI: `ModuleYamlParser` would stay unchanged and initialization logic from `ModuleYamlParserInstance` would be hidden in DI-container.

Reply:
But it can be created twice by mistake now.
It's ok, to move 30 lines of code to ModuleYamlParser, and will be easier to add new parsers (by modifying one file instead of two).
Also, it will be easier to use, ModuleYamlParser.Instance instead of ModuleYamlParserInstance.Get().

Reply:
Several existing instances of ModuleYamlParser is not a mistake, so there is no need to enforce that only one instance can be created.
Increasing file size by 1/3 will affect readability. The initialization logic doesn't belong near business-logic and is a subject to frequent change. Things that change often should be separated.

How come calling `ModuleYamlParser.Instance` is easier than `ModuleYamlParserInstance.Get()`? What is exactly "easier"?

Reply:
Ok, but should we rename it to `ModuleYamlParserProvider` of `_Factory`, because it is not instance of `ModuleYamlParser` and just creates and keep it?

Reply:
That makes sense.
Done: https://github.com/obiwan-smirnobi/cement/commit/58acc2c3f6c328d7d60b30db66339a0bf53906ee